src/daemon/ca/status.rs

```diff
@@ -109,37 +120,57 @@ impl StatusStore {
         entitlements: &Entitlements,
         next_run_seconds: i64,
     ) -> KrillResult<()> {
-        let _lock = self.lock.write().await;
-        let mut status = self.get_ca_status(ca)?;
-        status
-            .parents
-            .set_entitlements(parent, uri, entitlements, next_run_seconds);
-        self.set_ca_status(ca, &status)
+        self.update_ca_status(ca, |status| {
+            status
+                .parents
+                .set_entitlements(parent, uri, entitlements, next_run_seconds)
+        })
+        .await
     }
 
     pub async fn remove_parent(&self, ca: &Handle, parent: &ParentHandle) -> KrillResult<()> {
-        let _lock = self.lock.write().await;
-        let mut status = self.get_ca_status(ca)?;
-        status.parents.remove(parent);
-
-        Ok(())
+        self.update_ca_status(ca, |status| status.parents.remove(parent)).await
     }
 
-    pub async fn set_status_repo_failure(&self, ca: &Handle, uri: ServiceUri, error: &Error) -> KrillResult<()> {
-        let _lock = self.lock.write().await;
-        let mut status = self.get_ca_status(ca)?;
+    pub async fn set_child_success(
+        &self,
+        ca: &Handle,
+        child: &ChildHandle,
+        user_agent: Option<String>,
+    ) -> KrillResult<()> {
+        self.update_ca_child_status(ca, child, |status| status.set_success(user_agent))
+            .await
+    }
 
+    pub async fn set_child_failure(
+        &self,
+        ca: &Handle,
+        child: &ChildHandle,
+        user_agent: Option<String>,
+        error: &Error,
+    ) -> KrillResult<()> {
         let error_response = Self::error_to_error_res(&error);
 
-        status.repo.set_failure(uri, error_response);
-        self.set_ca_status(ca, &status)
+        self.update_ca_child_status(ca, child, |status| status.set_failure(user_agent, error_response))
+            .await
+    }
+
+    pub async fn get_children_status(&self, ca: &Handle) -> KrillResult<ChildrenConnectionStats> {
```

Comment:
The code mixes and matches the terms status and stats which are not strictly the same thing. Perhaps review the  places where these terms are used for consistency and correctness?

Reply:
Using 'stats' now for the most part. Right now this is essentially the last connection status, but it could be extended to include more information.